File: nodejs-tools/nodejs-compiler-tools/formatter-xml/src/format_comment_node.test.ts
import { test, expect } from 'vitest';
import { XmlCommentNode } from '@vighnesh153/parser-xml';
import { formatCommentNode } from './format_comment_node';
import { parseProgram } from './test_utils';

test('should format comment node with 0 indentation level', () => {
  const [parser, program] = parseProgram(
    `<!-- <uses-permission android:name="com.android.providers.tv.permission.WRITE_EPG_DATA" /> -->`
  );

  expect(parser.errors.length).toBe(0);
  expect(program.statements.length).toBe(1);
  expect(
    formatCommentNode({
      commentNode: program.statements[0] as XmlCommentNode,
      indentation: 4,
      indentationLevel: 0,
    })
  ).toMatchInlineSnapshot(
    `"<!-- <uses-permission android:name="com.android.providers.tv.permission.WRITE_EPG_DATA" /> -->"`
  );
});

test('should format comment node with 2 indentation level', () => {
  const [parser, program] = parseProgram(
    `<!-- <uses-permission android:name="com.android.providers.tv.permission.WRITE_EPG_DATA" /> -->`
  );

  expect(parser.errors.length).toBe(0);
  expect(program.statements.length).toBe(1);
  expect(
    formatCommentNode({
      commentNode: program.statements[0] as XmlCommentNode,
      indentation: 4,
      indentationLevel: 2,
    })
  ).toMatchInlineSnapshot(
    `"        <!-- <uses-permission android:name="com.android.providers.tv.permission.WRITE_EPG_DATA" /> -->"`
  );
});

test('should format comment node with 2 indentation level and 3 indentation', () => {
  const [parser, program] = parseProgram(
    `<!-- <uses-permission android:name="com.android.providers.tv.permission.WRITE_EPG_DATA" /> -->`
  );

  expect(parser.errors.length).toBe(0);
  expect(program.statements.length).toBe(1);
  expect(
    formatCommentNode({
      commentNode: program.statements[0] as XmlCommentNode,
      indentation: 3,
      indentationLevel: 2,
    })
  ).toMatchInlineSnapshot(
    `"      <!-- <uses-permission android:name="com.android.providers.tv.permission.WRITE_EPG_DATA" /> -->"`
  );
});

File: tools-jvm/XmlFormatter/src/test/kotlin/dev/vighnesh153/xml/formatter/fmt/FormatCommentNodeTest.kt
package dev.vighnesh153.xml.formatter.fmt

import dev.vighnesh153.xml.formatter.ast.XmlCommentNode
import dev.vighnesh153.xml.formatter.lexer.LexerInputReader
import dev.vighnesh153.xml.formatter.lexer.XmlLexer
import dev.vighnesh153.xml.formatter.parser.XmlParser
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

class FormatCommentNodeTest {
    private val lexerInputReader =
        LexerInputReader(
            input = """
            <!-- <uses-permission android:name="com.android.providers.tv.permission.WRITE_EPG_DATA" /> -->
        """.trimIndent()
        )
    private val lexer = XmlLexer(lexerInputReader)
    private val parser = XmlParser(lexer)
    private val program = parser.parseProgram()

    @Test
    fun formatsWithZeroIndentation() {
        assertEquals(0, parser.errors.size)
        assertEquals(1, program.statements.size)
        assertEquals(
            """<!-- <uses-permission android:name="com.android.providers.tv.permission.WRITE_EPG_DATA" /> -->""",
            (program.statements.first() as XmlCommentNode).format(
                indentationLevel = 0,
                indentation = 4,
            )
        )
    }

    @Test
    fun formatsWithTwoIndentationLevel() {
        assertEquals(0, parser.errors.size)
        assertEquals(1, program.statements.size)
        assertEquals(
            """        <!-- <uses-permission android:name="com.android.providers.tv.permission.WRITE_EPG_DATA" /> -->""",
            (program.statements.first() as XmlCommentNode).format(
                indentationLevel = 2,
                indentation = 4,
            )
        )
    }

    @Test
    fun formatsWithTwoIndentationLevelAndThreeIndentation() {
        assertEquals(0, parser.errors.size)
        assertEquals(1, program.statements.size)
        assertEquals(
            """      <!-- <uses-permission android:name="com.android.providers.tv.permission.WRITE_EPG_DATA" /> -->""",
            (program.statements.first() as XmlCommentNode).format(
                indentationLevel = 2,
                indentation = 3,
            )
        )
    }
}